addons/mass_mailing/static/src/interactions/subscribe.js

```diff
@@ -0,0 +1,336 @@
+     */
+    async onFeedbackClick() {
+        const formData = new FormData(
+            document.querySelector("div#o_mailing_subscription_feedback form")
+        );
+        const optoutReasonId = parseInt(formData.get("opt_out_reason_id"));
+        const result = await this.waitFor(
+            rpc("/mailing/feedback", {
+                csrf_token: formData.get("csrf_token"),
+                document_id: this.customerData.documentId,
+                email: this.customerData.email,
+                feedback: formData.get("feedback"),
+                hash_token: this.customerData.hashToken,
+                last_action: this.lastAction,
+                mailing_id: this.customerData.mailingId,
+                opt_out_reason_id: optoutReasonId,
+            })
+        );
+        this.protectSyncAfterAsync((result) => {
+            if (result === true) {
+                this.isFeedbackReadOnly = true;
+                this.setFeedbackInfoMessage("feedback_sent");
+            } else {
+                this.setFeedbackInfoMessage(result);
+            }
+            this.lastAction = result === true ? "feedback_sent" : result;
+        })(result);
+    }
+
+    /*
+     * Toggle feedback textarea display based on reason configuration
+     */
+    onOptOutReasonClick(ev) {
+        this.showFeedbackTextbox = ev.target.dataset["isFeedback"];
+        this.updateDisplay();
+    }
+
+    /**
+     * Parse start values of mailing lists subscriptions based on generated DOM
+     * from server. Done here to avoid having to generate it server-side and
+     * propagating it through various layers.
+     */
+    getListInfo() {
+        return [...document.querySelectorAll("#o_mailing_subscription_form_manage input")].map(
+            (node) => {
+                const listInfo = {
+                    id: parseInt(node.getAttribute("value")),
+                    member: node.dataset.member === "1",
+                    name: node.getAttribute("title"),
+                    opt_out: node.getAttribute("checked") !== "checked",
+                };
+                return listInfo;
+            }
+        );
+    }
+
+    onActionDone(callKey) {
+        this.lastAction = callKey;
+        this.updateDisplay();
+    }
+
+    /*
+     * Disable the mailing list selection form if necessary, reactivate the
+     * feedback form (which anyway stays hidden until necessary), empty the
+     * feedback textarea, and empty the feedback info text message.
+     */
+    updateDisplay() {
+        this.isFormReadonly = this.customerData.isBlocklisted;
+        this.isFeedbackReadOnly = false;
+        document.querySelector("div#o_mailing_subscription_feedback textarea").value = "";
+        document.getElementById("o_mailing_subscription_feedback_info").textContent = "";
+    }
+
+    /*
+     * Display feedback (visual tips) to the user concerning the last done action.
+     */
+    setUpdateInfoMessage(infoKey, template) {
```

Comment:
same here maybe "last done action" does not really help. Maybe rename this `setListSubscriptionResultMessage`

And/or similar comment for the body, could maybe use t-out and a `listSubscriptionUpdatedTemplate` where `null` is hidden (`showUpdateInfo` becomes redundant, can be derived as a getter) and otherwise you'd set the template wherever this is called instead.